src/NServiceBus.Core/Serialization/SerializationDefinition.cs

```diff
@@ -7,6 +7,36 @@
     /// </summary>
     public abstract class SerializationDefinition
     {
+        /// <inheritdoc />
+        bool Equals(SerializationDefinition other)
```

Comment:
i think this overridign equals and GetHashCode is wrong. people implement this and them overriding these members could have some very weird side effects. Since we only support on instance of a given type why not just cache on the type name (runtimetypehandle)